tao_compiler/mlir/disc/transforms/disc_lower_quantize_and_dequantize.cc

```diff
@@ -119,11 +119,12 @@ struct DequantizeOpConverter
     Value bcastedScale = rewriter.create<mhlo::DynamicBroadcastInDimOp>(
         loc, outTy, op.getScale(), inputShape, op.getAxis());
     auto zeroPointTy = op.getZeroPoint().getType().cast<RankedTensorType>();
-    auto bcastedInputOrZeroPointTy =
-        RankedTensorType::get(inputTy.getShape(), zeroPointTy.getElementType(),
-                              zeroPointTy.getEncoding());
+    auto bcastedInputOrZeroPointTy = RankedTensorType::get(
+        inputTy.getShape(), scaleTy.getElementType(), scaleTy.getEncoding());
+    Value castedZeroPoint =
+        rewriter.create<mhlo::ConvertOp>(loc, scaleTy, op.getZeroPoint());
     Value bcastedZeroPoint = rewriter.create<mhlo::DynamicBroadcastInDimOp>(
-        loc, bcastedInputOrZeroPointTy, op.getZeroPoint(), inputShape,
+        loc, bcastedInputOrZeroPointTy, castedZeroPoint, inputShape,
```

Comment:
I'm not quite sure if it's always safe to cast` input/zero_point `to float before we do `input-zeropoint`. What if the quantization is int32 instead of int8? maybe we can leave a comment here to describe the requirements of such change.

Reply:
In my point of view, float32 would be enough in most cases. Although this modification may bring risk, it can reduce large overhead meanwhile. Have added comments in case.